src/test/java/edu/cornell/kfs/concur/ConcurUtilsTest.java

```diff
@@ -1,99 +1,139 @@
-    
-    @Test
-    public void isBadTravelRequestURI(){      
-        Assert.assertFalse("The URI should have been an invalid Travel Request URI", ConcurUtils.isTravelRequestURI(BAD_TRAVEL_REQUEST_URI));
+
+    @ParameterizedTest
+    @MethodSource("testExtractCodeFromCodeAndDescriptionValueParams")
+    public void testExtractCodeFromCodeAndDescriptionValue(String codeAndDescription, String expectedValue) {
+        Assert.assertEquals(expectedValue, ConcurUtils.extractCodeFromCodeAndDescriptionValue(codeAndDescription));
     }
-    
-    @Test
-    public void isEmptyTravelRequestURI(){      
-        Assert.assertFalse("An empty URI should have been an invalid TravelRequest URI", ConcurUtils.isTravelRequestURI(StringUtils.EMPTY));
+
+    private static Stream<Arguments> testExtractCodeFromCodeAndDescriptionValueParams() {
+        return Stream.of(
+                Arguments.of(GOOD_CONCUR_FORMAT_CODE_AND_DESCRIPTION, KFS_FORMAT_ACCOUNT_NUMBER),
+                Arguments.of(StringUtils.EMPTY, StringUtils.EMPTY), 
+                Arguments.of(null, StringUtils.EMPTY));
     }
-    
-    @Test
-    public void extractAccountNumberFromConcurValue(){      
-        Assert.assertEquals(KFS_FORMAT_ACCOUNT_NUMBER, ConcurUtils.extractCodeFromCodeAndDescriptionValue(GOOD_CONCUR_FORMAT_CODE_AND_DESCRIPTION));
+
+    @ParameterizedTest
+    @MethodSource("testStringMatchesCodeAndDescriptionPatternParams")
+    public void testStringMatchesCodeAndDescriptionPattern(String patternToMatch, boolean expectedMatchStatus) {
+        Assert.assertEquals(expectedMatchStatus, ConcurUtils.stringMatchesCodeAndDescriptionPattern(patternToMatch));
+
     }
-    
-    @Test
-    public void extracCodeFromEmptyString(){      
-        Assert.assertEquals(StringUtils.EMPTY, ConcurUtils.extractCodeFromCodeAndDescriptionValue(StringUtils.EMPTY));
+
+    private static Stream<Arguments> testStringMatchesCodeAndDescriptionPatternParams() {
+        return Stream.of(
+                Arguments.of(VALUE_IN_CODE_AND_DESCRIPTION_FORMAT, true),
+                Arguments.of(VALUE_NOT_IN_CODE_AND_DESCRIPTION_FORMAT, false), 
+                Arguments.of(StringUtils.EMPTY, false));
     }
-    
+
     @Test
-    public void extracCodeFromNull(){      
-        Assert.assertEquals(StringUtils.EMPTY, ConcurUtils.extractCodeFromCodeAndDescriptionValue(null));
+    public void formatStringForErrorMessage() {
+        Assert.assertEquals(STRING_FORMATTED_FOR_ERROR_MESSAGE, ConcurUtils.formatStringForErrorMessage(
+                ConcurConstants.AccountingStringFieldNames.ACCOUNT_NUMBER, CHART, ACCOUNT_NUMBER));
     }
     
-    @Test
-    public void valueInCodeAndDescriptionFormat(){      
-        Assert.assertTrue("This value was expected to match the code and description pattern", ConcurUtils.stringMatchesCodeAndDescriptionPattern(VALUE_IN_CODE_AND_DESCRIPTION_FORMAT));
+    @ParameterizedTest
+    @MethodSource("testVuildValidationErrorMessageForWorkflowActionParams")
```

Comment:
Vuild => Valid (I think)

Reply:
ok

Reply:
I think there is one spot that still has testVuildValidationErrorMessageForWorkflowAction